week2/ShoppingList/cart.js

```diff
@@ -0,0 +1,289 @@
+    })
+
+    localStorage.setItem(ITEM_LIST_KEY, JSON.stringify(parsedItems));
+
+  }
+  else{ //체크 안되어 있으면(혹은 체크 해제했으면)
+    //모든 체크 버튼들 체크 해제
+    checkboxes.forEach((box)=>{
+      box.checked = false;
+      parsedItems[(box.parentElement.parentElement).id - 1].userCart = false;
+    })
+
+    localStorage.setItem(ITEM_LIST_KEY, JSON.stringify(parsedItems));
+
+  }
+}
+
+//모달 열고 닫는 로직 구성
+const allCheck = document.querySelector("#allCheck");
+allCheck.addEventListener("change", allCheckFunc)
+
+//buy버튼 누를 때 로직 구성
+const buyBtn = document.getElementById("buyBtn");
+buyBtn.addEventListener("click", (e)=> {
+  const buyModalWrapper = document.getElementById("buyModalWrapper");
+  buyModalWrapper.style.display = "flex"; /*이렇게 넣은 건 인라인스타일로 들어간다 */
+});
+
+//closeBuyModal 아이콘 누를 때 로직 구성
+const closeBuyModal = document.getElementById("closeBuyModal");
+closeBuyModal.addEventListener("click", ()=>{
+  const buyModalWrapper = document.getElementById("buyModalWrapper");
+  buyModalWrapper.style.display = "none"; 
+});
+
+
+//이제 모달만 구현하면 끝!!!!!!!!
+const modalContent = document.querySelector("#modalContent");
+function paintModalItem(obj)
+{  
+  //0. 아이템 틀 연결
+  const div = document.createElement("div") ;
+  div.classList.add("rowFlexDiv");
+  modalContent.appendChild(div);
+
+  // 1. 이미지 연결
+  const img = document.createElement("img");
+  img.setAttribute("src", obj.img);
+  div.appendChild(img);
+
+  // 2. 상품 카테고리 , 이름, 가격 설정 및 연결
+  const p1 = document.createElement("p");
+  p1.classList.add("cartItemP");
+  p1.innerText = `카테고리 : ${obj.category}`;
+
+  const p2 = document.createElement("p");
+  p2.classList.add("cartItemP");
+  p2.innerText = `상품명: ${obj.name}`;
+
+  const p3 = document.createElement("p");
+  //p3.classList.add("cartItemP");
+  p3.setAttribute("id", "cartItemPrice");
+  //toLocaleString(인자) 가 아니라 변환할 것.toLocaleString() 형식으로 사용해야한다
+  //메서드를 사용하는거라서 사용법이 그런 듯. 사용법 주의하고 정신차리기!
+  const localPrice = obj.price.toLocaleString(); 
+  p3.innerText = `가격 : ${localPrice}`;
+  
+  div.appendChild(p1);
+  div.appendChild(p2);
+  div.appendChild(p3);
+}
+
+ //그리고 총 가격 계산
+function deleteBuyList()
+{
+  while(modalContent.firstChild){ //만약 모달콘텐츠 박스 안에 첫번째 자식이 존재하면
+    (modalContent.firstChild).remove(); //다 없어질 때까지 차근 차근 요소 삭제
+    //혹은 modalContent.removeChild(modalContent.firstChild); 을 통해서도 삭제가 가능하다.
+  }
+}
+
+function renderBuyList() {
+  //매번 클릭시마다 paint 되도록 수정 - 새로운 문제 : 계속 껐다 키면 계속 렌더링 됨
+  //따라서 처음에 기존 렌더링 여부 확인 후, 렌더링 되어 있으면 삭제하는 로직 필요
+  deleteBuyList(); //필수!
+
+  const  NowAllObjs= JSON.parse(localStorage.getItem(ITEM_LIST_KEY));
+
+  const nowUserCart = NowAllObjs.filter((obj) => {
+    return (obj.userCart === true);
+  });
+  nowUserCart.forEach(paintModalItem);
+
+  //가격만 뽑아내어 새로운 배열 만듬
+  let allPrice = 0;
+  const priceArr = nowUserCart.map((obj) => obj.price);  
+  for (let aPrice of priceArr) //배열의 모든 요소 순회 (length를 사용할 수도 있음)
+  {
+    allPrice += aPrice;
+  }
```

Comment:
이 부분은 reduce 메소드를 사용해볼 수 있어요 !
[reduce 메소드란?](https://5kdk.tistory.com/2)
```suggestion
  const allPrice = nowUserCart.reduce((prev, curr) => prev + curr.price, 0);
```

Reply:
감사합니다! 덕분에 reduce 메서드에 대해서도 배울 수 있었습니다. 바로 적용해볼게요 😊